src/NuGet/Microsoft.Net.Compilers.Toolset/DesktopCompilerArtifacts.targets

```diff
@@ -67,8 +67,11 @@
       <_NoNGen Include="$(ArtifactsBinDir)csi\$(Configuration)\net472\System.Numerics.Vectors.dll"/>
       
       <DesktopCompilerArtifact Include="$(ArtifactsBinDir)csi\$(Configuration)\net472\System.*.dll" Exclude="@(_NoOptimizationData)" NgenArchitecture="all" NgenApplication="VBCSCompiler.exe" OverwriteNgenOptimizationData="true"/>
-      <DesktopCompilerArtifact Update="$(ArtifactsBinDir)csi\$(Configuration)\net472\System.Collections.Immutable.dll" NgenPriority="1" />
-      <DesktopCompilerArtifact Update="$(ArtifactsBinDir)csi\$(Configuration)\net472\System.Reflection.Metadata.dll" NgenPriority="1" />
+      
+      <!-- Note: do not use Update attribute (see https://github.com/microsoft/msbuild/issues/1124) -->
+      <DesktopCompilerArtifact NgenPriority="1" Condition="'%(Identity)' == '$(ArtifactsBinDir)csi\$(Configuration)\net472\System.Collections.Immutable.dll'" />
+      <DesktopCompilerArtifact NgenPriority="1" Condition="'%(Identity)' == '$(ArtifactsBinDir)csi\$(Configuration)\net472\System.Reflection.Metadata.dll'" />
```

Comment:
@tmat can you elaborate more on why this fixed the problem? I looked at the referenced issue in MSBuild and it's still not obvious to me what is going on here. My reading of this code is still that the `NgenPriority` attribute should only be updated for the cases where the `Condition` is true but based on the change that appears to be incorrect.

Reply:
The bug in msbuild is that the `Update` attribute is ignored. So the previous code was equivalent to `<DesktopCompilerArtifact NgenPriority="1"/>`, which to msbuild means set `NgenPriority` to `1` on all existing items `DesktopCompilerArtifact`. Adding the condition limits this to only those items whose identity is as specified, which is what we intended to do using the Update attribute.